src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -12,11 +18,37 @@ public MachineDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Cannot save machine " + machine, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Machine> criteriaQuery = criteriaBuilder.createQuery(Machine.class);
+            Root<Machine> root = criteriaQuery.from(Machine.class);
+            Predicate experience = criteriaBuilder.gt(root.get("age"), age);
+            criteriaQuery.select(root).where(experience);
```

Comment:
car experience? be careful with copy paste. If the the car has been manufactured in 2015 and I need to now whether it's older than 5 years. How would you calculate it?